### PR TITLE
464 datasets visible to logged in users with AUTHENTICATED status

### PR DESCRIPTION
**Pull request by tcnichol**
For this pull request, the following has been implemented.

If a dataset is public, all users can view the dataset, the files, and the metadata. 

A few things I might want to change:

In the route authorization.get_dataset_role, the response type is AuthorizationDB instead of just RoleType (which is the return type for similar methods for file, metadata, etc have.) So for the .get_dataset_role method I create an AuthorizationDB with RoleType 'Viewer' and return that if the dataset is public and the user has no role.

If anybody thinks it would be a better approach, I could change the response type of the .get_dataset_role method to RoleType. That seems like it might make more sense, but I wasn't sure if having the actual authorizationDB object was important anywhere else. 

I have also not figured out a solution for allowing users to view public datasets when not logged in. Thought it would be good to handle this, get some feedback, and then move forward.

### CODE REVIEW

**Comment by tcnichol** [comment]
This one is ready for review.
You can change a dataset to 'public' and all logged in viewers can see a public dataset, the files and metadata. 
Right now these are not visible if not logged in, and 'published' status exists but has no affect yet. Those will be later issues.

**Comment by longshuicy** [comment]
Based on yesterday's discussion, there seems to be more refactoring needed for this? @tcnichol

**Comment by tcnichol** [comment]
@longshuicy 

Yes. I'm going to try add new dependencies so that the public datasets will be visible when you are not logged in. So I'll keep working on this one.

I will also change this to draft.

File: frontend/src/routes.tsx [hunk 1]
```diff
@@ -21,6 +21,8 @@ import { RedirectLogin as RedirectLoginComponent } from "./components/auth/Redir
 import { RedirectLogout as RedirectLogoutComponent } from "./components/auth/RedirectLogout";
 import { Search } from "./components/search/Search";
 import { isAuthorized } from "./utils/common";
+import {isPublic} from "./utils/common"
+import {checkPublic} from "./utils/common"
```
[the remark this answers is not in the document]
Comment (tcnichol) [reply frontend/src/routes.tsx]: I took these out. Was trying to get this working for users when they were not logged in, but that was taking too long so I've limited this towards making datasets 'public' for users with accounts. 

I'll handle the other case with another issue.

**Comment by max-zilla** [comment]
the permissions generally seem to work as expected. however there are still a few buttons public (logged-in) users can see that they don't have permission to use: 
image
Can we hide the pencil edit icon and the delete button on the metadata tabs like we did for the Edit buttons for the dataset?

**Comment by tcnichol** [comment]
> the permissions generally seem to work as expected. however there are still a few buttons public (logged-in) users can see that they don't have permission to use:
> 
> (trying to edit someone else's metadata value) image Can we hide the pencil edit icon and the delete button on the metadata tabs like we did for the Edit buttons for the dataset?

Those buttons should no longer be visible if the user has viewer access only, so no add metadata, delete metadata, or edit.

**Review by lmarini — CHANGES_REQUESTED** [review]
Playing around with this. A couple of comments:

- I think Public should mean that anyone without an account could see the dataset. To the best of my understanding, right now the implementation makes it so the dataset is viewable to anyone who is logged in.
- When clicking on a file in a dataset that is public but is not shared with the current user, the app throws an error saying resource doesn't exist, which is probably because the file page doesn't respect this new toggle?
- Published is an option, but I don't believe it's being used. We should probably hide it / remove it until we implement it.

**Review by lmarini — CHANGES_REQUESTED** [review]
Overall looks good to me. A couple of proposed changes:

- [x] When a dataset is public and a user has access to it, they can execute an extractor. I am not sure we want to let them do this. We could disable/hide the button or maybe remove the extract tab all together?
- [ ] I don't think the "Published" option does anything right now. We should deleted from the list of options.

**Comment by lmarini** [comment]
@tcnichol can fix conflicts? Have you been able to address to the two small changes above? Thanks!

**Comment by tcnichol** [comment]
> @tcnichol can fix conflicts? Have you been able to address to the two small changes above? Thanks!

I'll be taking care of that this afternoon.

**Comment by tcnichol** [comment]
Pushed new commits, issues should be resolved and merge conflict fixed.

**Comment by tcnichol** [comment]
The status is now changed to AUTHENTICATED. The difference will be
AUTHENTICATED means any logged in user can view a dataset, but not someone who is not logged in. PUBLIC datasets can be seen whether you are logged in or not. A separate pull request will implement PUBLIC datasets.

**Review by lmarini — APPROVED** [review]
@tcnichol still works well after changes to label. I made one small change. I moved the submission button to bottom dialog actions in line with other popups and renamed to Update. Please take a look so we are consistent on all popups. Thank you.

**Pull request merged**